### PR TITLE
[ROCm] Use a 2D launch for the Triton bitmask kernel

### PR DESCRIPTION
**Pull request by AndreasKaratzas**
This PR changes only the ROCm path of `apply_token_bitmask_inplace_triton.py`. CUDA keeps the existing persistent launch. The current ROCm path uses a persistent 1D grid and loops over work with `tl.range(..., NUM_SMS)`. This patch switches ROCm to a regular 2D launch where each Triton program handles one `(vocab block, row)` pair. The motivation is an intermittent ROCm HSA exception we are seeing through vLLM structured-output inference. The failing vLLM logs consistently show this kernel being JITed shortly before the crash:

```text
Triton kernel JIT compilation during inference: apply_token_bitmask_inplace_kernel
HSA_STATUS_ERROR_EXCEPTION ... code: 0x1016
```

Filed an issue regarding this here:
- https://github.com/mlc-ai/xgrammar/issues/678

I could not reduce this to a standalone xgrammar crash. A direct xgrammar stress script using the same Qwen-shaped tensors did not fail locally. So I would not describe this as a proven xgrammar-only bug fix. The reason I think the PR still makes sense is that it removes the ROCm persistent-loop kernel shape from the suspect path, keeps the behavior unchanged, and did not show a performance regression on MI300.

I compared installed `xgrammar==0.2.1` against this patched source module across 480 GPU-timed cases. The benchmark measures only the in-place bitmask application after logits and bitmasks are already on GPU.

| Slice | Cases | Median delta vs installed |
|---|---:|---:|
| All cases | 480 | `-0.0039 ms` |
| Full-row cases | 300 | `-0.0039 ms` |
| Sparse-index cases | 180 | `-0.0039 ms` |
| Qwen vocab `151936` | 120 | `-0.0039 ms` |

On this MI300 setup, I do not see a steady-state performance regression. The patched ROCm path was slightly faster in this sweep, but the main takeaway is that the launch-shape change is behavior-preserving and performance-neutral for the cases tested.

### CODE REVIEW

**Comment by AndreasKaratzas** [comment]
i m going to be closing this one. There might be an issue with a particular triton version that vllm was running on.
